### ISSUE TITLE
Kernel crashes after any unhandled exception

### ISSUE THREAD

**Issue by giantamoeba** [ipython/ipykernel#370]
I have been experiencing this strange problem after updating to version 5.0 (5.1.0 still has the problem). All runs fine on version 4.9.

Minimal Example:
```python

import sys
from PyQt5.QtWidgets import QApplication

from qtconsole.rich_ipython_widget import RichJupyterWidget
from qtconsole.inprocess import QtInProcessKernelManager

app = QApplication(sys.argv)

kernel_manager = QtInProcessKernelManager()
kernel_manager.start_kernel()
kernel = kernel_manager.kernel
kernel.gui = 'qt'

kernel_client = kernel_manager.client()
kernel_client.start_channels()

widget = RichJupyterWidget()

widget.kernel_manager = kernel_manager
widget.kernel_client = kernel_client

widget.show()

sys.exit(app.exec_())
```
Then in the console, raise any exception:
```
In [1]: raise RuntimeError('ohai')
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-1-dba2c2e9b9ef> in <module>
----> 1 raise RuntimeError('ohai')

RuntimeError: ohai

In [2]: print()
ERROR: execution aborted
```

Has anyone else experienced this problem?

**Comment by gdementen** [comment]
yes, I have the same problem. Only workaround I found is downgrade back to 4.10

**Comment by giantamoeba** [comment]
I hope this is not the end of in-process kernels, I need in my applications a shared namespace so that the console can manipulate the application, and a kernel launched as a subprocess does not provide such a functionality..

**Comment by rayosborn** [comment]
It is vital to the [NeXpy application](http://nexpy.github.io/nexpy/) that this is resolved. I am trying to understand the new asynchronous message queues that were introduced in v5.0.0. It looks as if exceptions are not flushed from these queues before entering a new line in the terminal. I know that the main ipykernel developers do not consider the in-process kernel a high priority, although it is vital to a number of other applications, but if someone can explain where the message queues are reset in multi-kernel applications, that might help us to prepare a Pull Request to fix this issue.

**Comment by rayosborn** [comment]
I have submitted a Pull Request #376, which I think fixes this issue. It turned out to be as simple as adding a dummy `_abort_queues` function to the `InProcessKernel` that overrides the default function in the parent `Kernel` class. If you are able to test this fix, please let me know if still has any issues.

**Comment by rayosborn** [comment]
If you want to get your own code working before the next release of ipykernel (assuming the PR is accepted), it's fairly simple to monkey-patch the instance of the InProcessKernel class created by the kernel manager.
```
from qtconsole.inprocess import QtInProcessKernelManager

kernel_manager = QtInProcessKernelManager()
kernel_manager.start_kernel()
def _abort_queues(kernel):
    pass
kernel_manager.kernel._abort_queues = _abort_queues
```
I've tested that it fixes the issue in the NeXpy application.

**Comment by minrk** [comment]
Fixed by #376. Thanks @rayosborn!
